hub/core/dataset/dataset.py

```diff
@@ -1343,6 +1343,70 @@ def _set_read_only(self, value: bool, err: bool):
     def read_only(self, value: bool):
         self._set_read_only(value, True)
 
+    def cleanlab(
+        self,
+        module = None,
+        criterion = None,
+        optimizer = None,
+        optimizer_lr: int = 0.01,
+        device: str = "cpu",
+        epochs: int = 10,
+        folds: int = 5,
+        verbose: bool = True,
+        tensors: Optional[list] = None,
+        dataloader_train_params: [dict] = None,
+        dataloader_valid_params: Optional[dict] = None,
+        overwrite: bool = False
+        # skorch_kwargs: Optional[dict] = None,
+    ):
+        """
+        Cleans the labels of the dataset. Computes out-of-sample predictions and uses Cleanlab (github.com/cleanlab/cleanlab) open-source library
+        to automatically find label errors in machine learning datasets.Then, creates a set of tensors under label_issues group for the entire dataset.
+
+        Note:
+            Currently, only image classification task us supported. Therefore, the method accepts two tensors for the images and labels (e.g. ['images', 'labels']).
+            The tensors can be specified in dataloader_train_params or tensors. Any PyTorch module can be used as a classifier.
+
+        Args:
+            module (class): A PyTorch torch.nn.Module module (class or instance). In general, the uninstantiated class should be passed, although instantiated modules will also work. Default is torchvision.models.resnet18(), which is a PyTorch ResNet-18 model.
+            criterion (class): A PyTorch criterion. The uninitialized criterion (loss) used to optimize the module. Default is torch.nn.CrossEntropyLoss.
+            optimizer (class): A PyTorch optimizer. The uninitialized optimizer (update rule) used to optimize the module. Default is torch.optim.SGD.
+            optimizer_lr (int): The learning rate passed to the optimizer. Default is 0.01.
+            device (str): A PyTorch device. The device on which the module and criterion are located. Default is "cpu".
+            epochs (int): The number of epochs to train for each fit() call. Default is 10.
+            tensors (list): A list of tensor names that would be considered for cleaning (e.g. ['images', 'labels']).
+            dataloader_train_params (dict): Keyword arguments to pass into torch.utils.data.DataLoader. Options that may especially impact accuracy include: shuffle, batch_size.
```

Comment:
Not sure if this would be the best way to pass the parameters for the Hub's PyTorch Dataloader. Are there any other approaches that could make sense?